R/ts_plot.R

```diff
@@ -11,47 +11,39 @@
 #' @return ggplot Object
 #' @export
 #' @examples
-#' ts_plot("~/Desktop/ts4_jjshares.csv", "earnings", 4)
+#' time <- c("1950 Q1", "1950 Q2", "1950 Q3", "1950 Q4",
+#'           "1951 Q1", "1951 Q2", "1951 Q3", "1951 Q4")
+#' earnings <- c(0.71, 0.63, 0.82, 0.91, 0.71, 0.63, 0.82, 0.91)
+#' ts_data <- tibble::tibble(time, earnings)
+#' ts_plot(data=ts_data, col="earnings", frequency=4)
 ts_plot <- function(data, col, frequency){
-  
+
   # basic check of input
-  if(typeof(data) != 'character'){
-    stop("The path of the csv file should be a string")
-  }
-  
-  if(length(readr::read_csv(data)) == 1){
-    stop("The type of the input data must be a csv file")
+  if (!class(data)[1] %in% c("tbl_df", "tbl", "data.frame")) {
+    stop("Data must be of type tibble or data.frame")
   }
-  
-  
+
+
   if(typeof(col) != 'character') {
     stop("The column name should be a string")
   }
-  
+
   if(typeof(frequency) != "double") {
     stop("The frequency of time series should be an integer chosen from 1, 4, 12, 52")
   }
-  
+
   if(frequency %in% c(1, 4, 12, 52) == FALSE) {
     stop("The frequency of time series should be an integer chosen from 1, 4, 12, 52")
   }
-  
-  # read the data into a dataframe
-  df <- readr::read_csv(data)
-  
+
   # check the input further
-  if (c(col) %in% colnames(df) == FALSE){
+  if (c(col) %in% colnames(data) == FALSE){
     stop("The column names were not found")
   }
-  
+
   # convert csv file into a time series object
```

Comment:
This function should no longer take a csv file. This comment is inconsistent with the rest of the function.